Add throttling to download progress events

KleggerKoder opened the pull request:
<!--
Please make sure the checklist boxes are all checked before submitting the PR. The checklist is intended as a quick reference, for complete details please see our Contributor Guidelines:

http://cordova.apache.org/contribute/contribute_guidelines.html

Thanks!
-->

### Platforms affected

iOS and Android

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
Way too many progress events being fired and makes all the application side progress indicators bounce around and sometimes crash the app.  I believe download speeds may have affected the number of updates as well.


### Description
<!-- Describe your changes in detail -->
Added some throttling to reduce message rate to 1/100th what it use to.


### Testing
<!-- Please describe in detail how you tested your changes. -->
1/100th seemed reasonable for downloading a variety of small and large files on fast connections.
Your experience may vary depending on connection speeds.


### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

KleggerKoder committed:
Add some hacky throttling to the progress event

KleggerKoder committed:
Fine tune throttle loop

KleggerKoder committed:
Add progress throttling to iOS

KleggerKoder committed:
Add throttling to iOS

ProudOne committed:
Merge branch 'master' of https://github.com/apache/cordova-plugin-file-transfer into apache-master

# Conflicts:
#	src/android/FileTransfer.java

wilk-polarny commented:
Hi, I would like to suggest a different approach, since fixing the underlying Cordova issue is arduous.
Rather than always skipping 100 elements, I would suggest to:

a) Only emit computable events
b) compute and track the progress and only emit an event if the current progress changed significantly. For example, computer the progress %, and only emit an event every 1% - or once a certain amount of transferred bytes has been reached - based on the content length.

ProudOne committed:
Merge branch 'apache:master' into master

GitToTheHub committed:
iOS: Cast to `CDVFile` when getting the plugin by `getCommandInstance`

- `CDVCommandDelegate getCommandInstance` returns a `CDVPlugin` which does not know about the selectors of an inherited class like `CDVFile`. If not casted, XCode will throw an error like `error: no visible @interface for 'CDVPlugin' declares the selector 'filesystemForURL:'`.

ProudOne committed:
Merge branch 'apache:master' into master

ProudOne committed:
Merge pull request #3 from GitToTheHub/pr-ios-cdvfile-cast

iOS: Cast to `CDVFile` when getting the plugin by `getCommandInstance`
